plugins/MySQL.cpp

```diff
@@ -81,6 +81,33 @@ string MySQLPacket::serializeHandshake() {
     SAppend(handshake.payload, &connectionID, 4); // connection_id
     handshake.payload += (string) "xxxxxxxx";     // auth_plugin_data_part_1
     handshake.payload += lenEncInt(0);            // filler
+
+    static const unsigned CLIENT_LONG_PASSWORD = 0x00000001;
+    static const unsigned CLIENT_PLUGIN_AUTH   = 0x00080000;
+    const unsigned capability_flags = CLIENT_LONG_PASSWORD | CLIENT_PLUGIN_AUTH;
+
+    const unsigned short capability_flags_1 = (const unsigned short)(capability_flags);
+    const unsigned short capability_flags_2 = (const unsigned short)(capability_flags >> 16);
+    SAppend(handshake.payload, &capability_flags_1, 2); // capability_flags_1 (low 2 bytes)
```

Comment:
@quinthar I don't really understand how `lenEncInt()` is supposed to work for a fixed length integer. Say protocol requires 2 bytes int of `capability_flags` and these flags just happens to have a value < 256. `lenEncInt()` as far as I understand would just encode it as one byte which would violate the protocol?